src/service/gif.js

```diff
@@ -0,0 +1,16 @@
+import api from './config';
+
+export const getGifFeatured = async () =>
+  new Promise((resolve, reject) => {
+    api
+      .get('/chat/gif/featured')
+      .then((res) => {
+        resolve(res.data);
+      })
+      .catch((err) => {
+        if (__DEV__) {
+          console.log('get gif featured: ', err);
+        }
+        reject(err);
+      });
+  });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

While the code is generally fine, it's not a good practice to use `console.log` for error logging even in development mode. It would be better to use a proper logging library that can provide more context and control over log levels. Also, you should consider handling the error in a way that doesn't disrupt the user experience.

```diff
-        if (__DEV__) {
-          console.log('get gif featured: ', err);
-        }
+        Logger.error('getGifFeatured', err);
```
Also, it's recommended to handle HTTP errors based on their status codes. For instance, if the server returns a 404 (Not Found), you might want to handle this differently than a 500 (Internal Server Error). You could add a switch case or if-else ladder to handle different HTTP status codes. 

Remember, unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code. So, always make sure to handle all possible outcomes of a promise.


<!-- This is an auto-generated comment by OSS CodeRabbit -->